cura/Settings/MachineManager.py

```diff
@@ -1531,3 +1531,31 @@ def _updateUponMaterialMetadataChange(self) -> None:
         with postponeSignals(*self._getContainerChangedSignals(), compress = CompressTechnique.CompressPerParameterValue):
             self.updateMaterialWithVariant(None)
             self._updateQualityWithMaterial()
+
+    @pyqtSlot(QObject)
+    def calculateSettingValue(self, propertyProvider):
```

Comment:
This is also not good because now `MachineManager` depends on the PropertyProviders which are UI modules. It will be very difficult to separate the core functionality out after this.